src/sentry/models/integrations/external_actor.py

```diff
@@ -48,11 +48,18 @@ def delete(self, **kwargs):
 
 
 def process_resource_change(instance, **kwargs):
+    from sentry.models import Organization, Project
     from sentry.tasks.codeowners import update_code_owners_schema
 
-    return update_code_owners_schema.apply_async(
-        kwargs={"organization": instance.organization, "integration": instance.integration}
-    )
+    def _spawn_task():
+        try:
+            update_code_owners_schema.apply_async(
+                kwargs={"organization": instance.organization, "integration": instance.integration}
```

Comment:
Yes absolutely. The task signature here is problematic for many reasons but I don't want to change it right now.

Reply:
The problems I'm aware of are:

1. The instance that goes into the queue message could be minutes or hours old when the task is processed. This can result in stale reads.
2. The instances are pickled into the queue messages. This increases the amount of memory that rabbit uses to maintain queues. The additional data also adds network overhead as the messages have to go back and forth between consumers and rabbit.

Reply:
Other problems:
 - If you create a task, then a deploy goes out that renames or moves the model, then the task will fail to unpickle the model when it attempts to run.
 - Similarly there can be issues if you add new fields to a model and attempt to unpickle, and access the new field.

Reply:
yes, all of the above + we're adding a unnecessary database query here that runs synchronously and increases response times across the board. i saw in tests that `instance.organization` is accessed the first time here.